File: deadfish.py
def parse(data: str) -> list:
    result: list = []
    value = 0

    for letter in data:
        if letter == "i":
            value += 1
        elif letter == "d":
            value -= 1
        elif letter == "s":
            value *= value
        elif letter == "o":
            result.append(value)

    return result

def main():
    tests = ["iiisdoso", "ooo", "ioioio", "codewars", "isoisoiso", "ssdizbsnsdkkincuybidds"]
    expected = [[8,64], [0,0,0], [1,2,3], [0], [1,4,25], None] 
    for i, par in enumerate(tests):
        got = parse(par)
        print(got, expected[i])
        print(got == expected[i])

if __name__ == "__main__":
    main()

File: deadfish.go
package main

import "fmt"

func Parse(data string) []int {
	var result []int
	var value int
	for _, letter := range data {
		switch string(letter) {
		case "i":
			value += 1
		case "d":
			value += -1
		case "s":
			value *= value
		case "o":
			result = append(result, value)
		}
	}

	if result == nil {
		return nil
	}

	return result
}

func main() {
	tests := []string{"iiisdoso", "ooo", "ioioio", "codewars", "isoisoiso", "ssdizbsnsdkkincuybidds"}
	expected := [][]int{[]int{8, 64}, []int{0, 0, 0}, []int{1, 2, 3}, []int{0}, []int{1, 4, 25}, nil}

	for i, par := range tests {
		got := Parse(par)
		fmt.Println(got, expected[i])
	}
}
